vms/avm/service.go

```diff
@@ -157,19 +160,24 @@ func (service *Service) GetUTXOs(r *http.Request, args *GetUTXOsArgs, reply *Get
 	}
 
 	chainID := ids.ID{}
+	if args.SourceChain == "" {
+		chainID = service.vm.ctx.ChainID
+	} else {
+		id, err := service.vm.ctx.BCLookup.Lookup(args.SourceChain)
+		if err != nil {
+			return fmt.Errorf("problem parsing chainID %q: %w", args.SourceChain, err)
+		}
+		chainID = id
+	}
 
 	addrSet := ids.ShortSet{}
 	for _, addrStr := range args.Addresses {
 		addrChainID, addr, err := service.vm.ParseAddress(addrStr)
 		if err != nil {
 			return fmt.Errorf("problem parsing address %q: %w", addrStr, err)
 		}
-		if chainID.IsZero() {
-			chainID = addrChainID
-		}
-		if !chainID.Equals(addrChainID) {
-			return fmt.Errorf("addresses from multiple chains provided: %q and %q",
-				chainID, addrChainID)
+		if !addrChainID.Equals(service.vm.ctx.ChainID) {
+			return fmt.Errorf("address %s was for the incorrect chain: %s", addrStr, addrChainID)
```

Comment:
I would change this to:
`return fmt.Errorf("address on chain %s given but should be on chain %s", addrStr, addrChainID)`

Reply:
I'm going to change to the following `return fmt.Errorf("address %s is on chain %s but should be on chain %s", addrStr, addrChainID, service.vm.ctx.ChainID)` because the intention here is that addresses passed in will always be the address that the atomic UTXOs have been exported to